### PR TITLE
SeatGeek Sports - new IA

### PR DESCRIPTION
**Pull request by MariagraziaAlastra**
//cc @jagtalon @abeyang
This is a WIP, I still need to add the test files, test throughly cross-browser compatibility, ecc.
Anyway, template:

**What does your Instant Answer do?**
Returns upcoming sport events for a given team/performer (or also big sport events like Miami Heat).

**What problem does your Instant Answer solve (Why is it better than organic links)?**
It's faster than the organic links. It lets users easily see what are the upcoming sport events
for a given team or performer.

**What is the data source for your Instant Answer? (Provide a link if possible)**
[SeatGeek](https://seatgeek.com/)

**Why did you choose this data source?**
Because it's exhaustive, has a good API and we're already using it for concert events.

**Are there any other alternative (better) data sources?**
Don't think so

**What are some example queries that trigger this Instant Answer?**
"upcoming matches Milan", "events UEFA champions league", etc

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**
People who loves to watch sports, especially live!

**Is this Instant Answer connected to a DuckDuckHack [Instant Answer idea](https://duck.co/ideas)?**
There's an Asana task for it!

**Which existing Instant Answers will this one supersede/overlap with?**
No one

**Are you having any problems? Do you need our help with anything?**
Not so far :) 

**What are the terms of use for the API? Will DuckDuckGo need specific authorization (e.g. an API key)? Are there any costs associated with API usage?**
It's free!

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**
screen shot 2014-11-02 at 20 38 07
## Checklist

Please place an 'X' where appropriate.

```
[x] Added metadata and attribution information
[] Wrote test file and added to t/ directory
[x] Verified that instant answer adheres to design guidelines (https://duck.co/duckduckhack/design_styleguide)
[x] Verified that instant answer adheres to code styleguide (https://duck.co/duckduckhack/code_styleguide)
[x] Tested cross-browser compatibility

    Please let us know which browsers/devices you've tested on:
    - Windows 8
        [] Google Chrome
        [] Firefox
        [] Opera
        [] IE 10

    - Windows 7
        [x] Google Chrome
        [x] Firefox
        [] Opera
        [] IE 8
        [] IE 9
        [] IE 10

    - Windows XP
        [] IE 7
        [] IE 8

    - Mac OSX
        [] Google Chrome
        [] Firefox
        [] Opera
        [] Safari

    - iOS (iPhone)
        [] Safari Mobile
        [] Google Chrome
        [] Opera

    - iOS (iPad)
        [] Safari Mobile
        [] Google Chrome
        [] Opera

    - Android
        [] Firefox
        [] Native Browser
        [] Google Chrome
        [] Opera
```

### CODE REVIEW

File: lib/DDG/Spice/SeatGeek/Sports.pm [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+topics "entertainment";
+attribution github => ['https://github.com/MariagraziaAlastra','MariagraziaAlastra'];
+
+triggers startend => 'upcoming matches', 'events', 'event', 'upcoming match', 'matches', 'sports', 'schedule';
+
+spice to => 'http://api.seatgeek.com/2/events?q=$1&taxonomies.name=sports&callback={{callback}}';
+
+handle remainder_lc => sub {
+    # Removes triggers from the query
+    $_ =~ s/^(:?(upcoming\s*)?(match(es)?))|(events?)|(sports)|(schedule)$//gi;
```
Comment (jagtalon) [inline lib/DDG/Spice/SeatGeek/Sports.pm]: Won't `remainder_lc` remove it for us?

File: share/spice/seat_geek/sports/seat_geek_sports.js [hunk 2]
```diff
@@ -0,0 +1,128 @@
+(function(env) {
+    "use strict";
+    env.ddg_spice_seat_geek_sports = function(api_result) {
+
+        if(!api_result || api_result.error || api_result.events.length === 0) {
+            return Spice.failed('seat_geek_sports');
+        }
+
+        var query = DDG.get_query();
+        var clean_query = query.replace(/((upcoming\s)?(match(es)?))|(events?)|(schedule)|(sports)/, '').trim().toLowerCase();
```
Comment (jagtalon) [inline share/spice/seat_geek/sports/seat_geek_sports.js]: `.trim()` won't work on IE 8 unfortunately :/

File: share/spice/seat_geek/sports/seat_geek_sports.js [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+                    }
+
+                    return;
+                }
+
+                function getDay(date) {
+                    if(date) {
+                        var day = date.getDate();
+                        var suffix = "";
+                        switch (day) {
```
Comment (jagtalon) [inline share/spice/seat_geek/sports/seat_geek_sports.js]: You might find `DDG.getOrdinal` useful!

```javascript
> DDG.getOrdinal(10)
"10th"
```

**Comment by jagtalon** [comment]
@MariagraziaAlastra the text can overflow on mobile devices:

screen shot 2015-01-13 at 3 16 00 pm

Maybe you can add the `one-line` class on that element?

File: share/spice/seat_geek/sports/seat_geek_sports.js (hunk 3 above)
Comment (MariagraziaAlastra) [inline share/spice/seat_geek/sports/seat_geek_sports.js]: @jagtalon Thanks, didn't know about that one!

**Comment by MariagraziaAlastra** [comment]
@jag Ready for more testing! It's live on my instance :)

Desktop:
screen shot 2015-01-15 at 16 12 49

Mobile:
screen shot 2015-01-15 at 16 23 24

**Comment by jagtalon** [comment]
@MariagraziaAlastra that. looks. sweet.

**Comment by jagtalon** [comment]
@MariagraziaAlastra Haven't checked the code yet, but are there bigger images available? It's blurry on high-resolution screens:

img_0511

You can use something like this to pick the right image: https://github.com/duckduckgo/zeroclickinfo-spice/blob/master/share/spice/currency/currency.js#L89

**Comment by MariagraziaAlastra** [comment]
@jagtalon We've got just there so far!

**Comment by abeyang** [comment]
@jagtalon yeah, those images look pretty bad -- I'll look into them. In the meantime, I'll be posting some design comments for you, @MariagraziaAlastra !

**Comment by abeyang** [comment]
pasted_image_1_15_15__10_12_am

**Comment by abeyang** [comment]
Here are some css changes:

```css
.tile, .tile--s, .tile--info, .tile--map {
font-size: 1em;
width: 16em;
}
.zci--seat_geek_sports .tile__title {
font-size: 1.357em;
line-height: 1.11;
padding-right: 40px;
}
```

Also, currently there is a 62% width set for .zci--seat_geek_sports .tile__title -- let's delete it:

screen shot 2015-01-15 at 10 51 14 am

**Comment by abeyang** [comment]
Finally--

pasted_image_1_15_15__10_53_am

**Comment by MariagraziaAlastra** [comment]
@abeyang @jagtalon changes made!
Desktop:
screen shot 2015-01-15 at 21 57 00

Mobile:
screen shot 2015-01-15 at 22 02 17

File: share/spice/seat_geek/sports/seat_geek_sports.js (hunk 2 above)
Comment (moollaza) [inline share/spice/seat_geek/sports/seat_geek_sports.js]: Just FYI guys, we have a polyfill for `trim` in our JS code so it's actually OK to use it. Regardless, it feels weird to use something that technically isn't supported, so I'm all for using `$.trim`

**Comment by jagtalon** [comment]
@MariagraziaAlastra 

- Date is missing in the dark theme. Adding the class `tx-clr--dk` for it should fix it.
- @abeyang maybe the assets should be transparent? It looks weird in the black theme.

screen shot 2015-01-19 at 3 21 39 pm

- Icons are cut-off on mobile:

screen shot 2015-01-19 at 3 24 51 pm
screen shot 2015-01-19 at 3 25 08 pm

**Comment by jagtalon** [comment]
@abeyang about https://github.com/duckduckgo/zeroclickinfo-spice/pull/1265#issuecomment-70131834, hi-res icons would be sweet! (☞ﾟヮﾟ)☞

**Comment by MariagraziaAlastra** [comment]
@jagtalon Changes made:

screen shot 2015-01-19 at 21 51 37
screen shot 2015-01-19 at 21 51 19

**Comment by jagtalon** [comment]
@MariagraziaAlastra nice!

**Comment by jagtalon** [comment]
@MariagraziaAlastra did another pass on it--I think this is awesome. The flags are still blurry on retina screens, but passing to @abeyang for a final design pass.

**Comment by MariagraziaAlastra** [comment]
Update:
schermata 2015-03-02 alle 17 50 07

**Comment by jagtalon** [comment]
Nice! Love that we're using the new templates.

**Comment by jagtalon** [comment]
@MariagraziaAlastra would this benefit from moment.js https://github.com/duckduckgo/zeroclickinfo-spice/issues/1554? Might be hard to refactor though, but it's there if you need it. :)

**Comment by jagtalon** [comment]
:+1: 

53hgk

**Comment by MariagraziaAlastra** [comment]
:cactus:

**Comment by moollaza** [comment]
@MariagraziaAlastra @jagtalon it's live! https://duckduckgo.com/?q=concerts+near+me&ia=concerts

**Pull request merged**